diff --git a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
index 8ea996d7ce68e007cce5f2b2ce2722b6aae704a7..78149b3ed17990c03014c2a1552f119c4f00614c 100644
--- a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
+++ b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
@@ -557,6 +557,7 @@ public abstract class ChromeFeatureList {
     public static final String WEB_APK_INSTALL_SERVICE = "WebApkInstallService";
     public static final String XSURFACE_METRICS_REPORTING = "XsurfaceMetricsReporting";
     public static final String WEB_OTP_CROSS_DEVICE_SIMPLE_STRING = "WebOtpCrossDeviceSimpleString";
+    public static final String BOOKMARKS_EXPORT_USESAF = "BookmarksExportUseSaf";
 
     @NativeMethods
     interface Natives {
