diff --git a/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/dialogs/DownloadLocationCustomView.java b/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/dialogs/DownloadLocationCustomView.java
index 8abf7ed59880344832205454e3aca2fb868c060e..23e6b42bf0661ab4efbd76afe17d911f8571ca4f 100644
--- a/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/dialogs/DownloadLocationCustomView.java
+++ b/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/dialogs/DownloadLocationCustomView.java
@@ -49,7 +49,7 @@ public class DownloadLocationCustomView
     private TextView mFileSize;
     private Spinner mFileLocation;
     private TextView mLocationAvailableSpace;
-    private CheckBox mDontShowAgain;
+    public CheckBox mDontShowAgain;
     private @DownloadLocationDialogType int mDialogType;
     private long mTotalBytes;
 
@@ -72,14 +72,14 @@ public class DownloadLocationCustomView
         mDontShowAgain = findViewById(R.id.show_again_checkbox);
     }
 
-    void initialize(@DownloadLocationDialogType int dialogType, long totalBytes) {
+    public void initialize(@DownloadLocationDialogType int dialogType, long totalBytes) {
         // TODO(xingliu): Remove this function, currently used by smart suggestion.
         mDialogType = dialogType;
         mTotalBytes = totalBytes;
         mDirectoryAdapter.update();
     }
 
-    void setTitle(CharSequence title) {
+    public void setTitle(CharSequence title) {
         mTitle.setText(title);
     }
 
@@ -87,7 +87,7 @@ public class DownloadLocationCustomView
         mSubtitleView.setText(subtitle);
     }
 
-    void setFileName(CharSequence fileName) {
+    public void setFileName(CharSequence fileName) {
         mFileName.setText(fileName);
     }
 
@@ -125,7 +125,7 @@ public class DownloadLocationCustomView
      * @return  The text that the user inputted as the name of the file.
      */
     @Nullable
-    String getFileName() {
+    public String getFileName() {
         if (mFileName == null || mFileName.getText() == null) return null;
         return mFileName.getText().toString();
     }
@@ -134,7 +134,7 @@ public class DownloadLocationCustomView
      * @return  The file path based on what the user selected as the location of the file.
      */
     @Nullable
-    DirectoryOption getDirectoryOption() {
+    public DirectoryOption getDirectoryOption() {
         if (mFileLocation == null) return null;
         DirectoryOption selected = (DirectoryOption) mFileLocation.getSelectedItem();
         return selected;
