diff --git a/chrome/android/java/src/org/chromium/chrome/browser/native_page/NativePageFactory.java b/chrome/android/java/src/org/chromium/chrome/browser/native_page/NativePageFactory.java
index 153e51705b9a239121fb9a13887c3692925f4973..3dec9ed3099d391023411bc4a11382bf14772e5b 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/native_page/NativePageFactory.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/native_page/NativePageFactory.java
@@ -15,6 +15,7 @@ import org.chromium.base.jank_tracker.JankTracker;
 import org.chromium.base.supplier.BooleanSupplier;
 import org.chromium.base.supplier.DestroyableObservableSupplier;
 import org.chromium.base.supplier.Supplier;
+import org.chromium.chrome.browser.app.ChromeActivity;
 import org.chromium.chrome.browser.app.download.home.DownloadPage;
 import org.chromium.chrome.browser.bookmarks.BookmarkPage;
 import org.chromium.chrome.browser.browser_controls.BrowserControlsMarginSupplier;
@@ -54,7 +55,7 @@ import org.chromium.ui.util.ColorUtils;
  * Creates NativePage objects to show chrome-native:// URLs using the native Android view system.
  */
 public class NativePageFactory {
-    private final Activity mActivity;
+    private final ChromeActivity mActivity;
     private final BottomSheetController mBottomSheetController;
     private final BrowserControlsManager mBrowserControlsManager;
     private final Supplier<Tab> mCurrentTabSupplier;
@@ -71,7 +72,7 @@ public class NativePageFactory {
 
     private NativePageBuilder mNativePageBuilder;
 
-    public NativePageFactory(@NonNull Activity activity,
+    public NativePageFactory(@NonNull ChromeActivity activity,
             @NonNull BottomSheetController sheetController,
             @NonNull BrowserControlsManager browserControlsManager,
             @NonNull Supplier<Tab> currentTabSupplier,
@@ -119,7 +120,7 @@ public class NativePageFactory {
 
     @VisibleForTesting
     static class NativePageBuilder {
-        private final Activity mActivity;
+        private final ChromeActivity mActivity;
         private final BottomSheetController mBottomSheetController;
         private final Supplier<NewTabPageUma> mUma;
         private final BrowserControlsManager mBrowserControlsManager;
@@ -132,7 +133,7 @@ public class NativePageFactory {
         private final JankTracker mJankTracker;
         private final Supplier<Toolbar> mToolbarSupplier;
 
-        public NativePageBuilder(Activity activity, Supplier<NewTabPageUma> uma,
+        public NativePageBuilder(ChromeActivity activity, Supplier<NewTabPageUma> uma,
                 BottomSheetController sheetController,
                 BrowserControlsManager browserControlsManager, Supplier<Tab> currentTabSupplier,
                 Supplier<SnackbarManager> snackbarManagerSupplier,
@@ -169,7 +170,7 @@ public class NativePageFactory {
         protected NativePage buildBookmarksPage(Tab tab) {
             return new BookmarkPage(mActivity.getComponentName(), mSnackbarManagerSupplier.get(),
                     mTabModelSelector.isIncognitoSelected(),
-                    new TabShim(tab, mBrowserControlsManager, mTabModelSelector));
+                    new TabShim(tab, mBrowserControlsManager, mTabModelSelector), mActivity);
         }
 
         protected NativePage buildDownloadsPage(Tab tab) {
